packages/flame/lib/src/extensions/rect.dart

```diff
@@ -52,6 +53,22 @@ extension RectExtension on Rect {
     ];
   }
 
+  /// Transform Rect using the transformation defined by [matrix].
+  ///
+  /// **Note:** Rotation matrices will increase the size of the [Rect] but they
+  /// will not rotate it as [Rect] does not have any rotational values.
+  ///
+  /// **Note:** Only non-negative scale transforms are allowed, if a negative
+  /// scale is applied it will return a zero-based [Rect].
+  ///
+  /// **Note:** The transformation will always happen from the center of the
+  /// `Rect`.
+  Rect transform(Matrix4 matrix) {
+    final topLeft = matrix.transform2(this.topLeft.toVector2());
+    final bottomRight = matrix.transform2(this.bottomRight.toVector2());
```

Comment:
Is it possible to calculate these values without creating new vectors? Since `transform2` also creates a new vector there will be four new vectors for every call to this method.

Reply:
I can do it without creating any vectors at all!